src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
+        // Create a new empty HashMap for the collection
+        let collection: HashMap<String, Hash> = HashMap::new();
     
-        // Serialize the collection to CBOR
+        // Serialize the HashMap to CBOR
         let cbor_data = to_vec(&collection)?;
-
-        // Log: Writing CBOR data to a temporary file
-        println!("Writing CBOR data to a temporary file...");
-        
-        // Create a temporary file to store the CBOR data
+    
+        // Write the CBOR data to a temporary file
         let temp_path = self.base_dir.join(format!("{}_collection.cbor", collection_name));
         std::fs::write(&temp_path, &cbor_data)?;
-        
-        // Convert the path to an absolute path
+    
+        // Convert the path to an absolute path and upload it to the store
         let absolute_temp_path = std::fs::canonicalize(&temp_path)?;
-
-        // Log: Uploading the CBOR data from file to the store
-        println!("Uploading the CBOR data from file to the store...");
-
-        // Upload the CBOR data to the store using upload_from_path
         let collection_hash = self.upload_from_path(absolute_temp_path).await?;
-
-        // Log: Storing the collection hash in the HashMap
-        println!("Storing the collection hash in the HashMap...");
-
-        // Store the collection hash in the HashMap
-        {
-            let mut map = self.collection_hashes.write().await;
-            map.insert(collection_name, collection_hash);
-        }
-
-        // Log: Collection created successfully
+    
+        // Store the new collection hash using the tag system
+        self.store.set_tag(
+            collection_name.to_string().into(),
+            Some(HashAndFormat::new(collection_hash, BlobFormat::Raw)),
+        ).await?;
+    
         println!("Collection created successfully with hash: {:?}", collection_hash);
-
         Ok(collection_hash)
     }
     
-
-    pub async fn set_file(&self, collection_name: String, path: String, hash: Hash) -> Result<Hash> {
-        // Retrieve the collection's current hash
-        let collection_hash = self.collection_hash(&collection_name).await?;
-    
-        // Read the collection data from the store
-        let collection_data = self.read_bytes(collection_hash).await?;
-        let mut collection: Collection = from_slice(&collection_data)?;
     
-        // Add or update the file in the collection
-        collection.files.insert(path, hash);
     
-        // Serialize the updated collection to CBOR
+    pub async fn set_file(&self, collection_name: &String, path: &String, file_hash: &Hash) -> Result<Hash> {
+         // Retrieve the current collection hash from the tag
+        println!("Attempting to retrieve tag for collection: {}", collection_name);
+
+        // Retrieve the collection hash using `collection_hash` (use this to get the collection)
+        let collection_hash = match self.collection_hash(&collection_name).await {
```

Comment:
Use `?` here for readability, we can use whatever error `self.collection_hash` returns as is